resolver.go

```diff
@@ -151,22 +158,36 @@ func setCommonFlags(msg *dns.Msg) {
 	msg.RecursionAvailable = true
 }
 
+func shuffleAddr(addr []net.IP) []net.IP {
```

Comment:
@bboreham It wasn't a typo. rfc 3484 lays out the rules for ordering the set of IPs when there is a mix of V4 and V6 addresses in different scopes. But for the IPs of a given type the sort is stable; ie: getaddrinfo() returns the v4 or v6 addresses in the same order it received from the upstream server. Hence the application has to pick one IP randomly to get the LB effect. Doing the shuffling in the embedded server helps the apps to get the LB benefit even if they are not coded to randomly pick one IP.